test/aws/mzcompose.py

```diff
@@ -0,0 +1,235 @@
+        caller = self.sts.get_caller_identity()
+        self.account_id = caller["Account"]
+        self.materialized_principal = caller["Arn"]
+
+
+def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
+    # Sleeping to wait for IAM to propagate is ugly and somewhat flaky, but
+    # there isn't an obviously better solution. This only runs in the nightly
+    # pipeline, so flakes are more tolerable than they would be if this ran in
+    # the PR pipeline.
+    parser.add_argument(
+        "--iam-propagation-seconds",
+        type=int,
+        default=10,
+        help="how long to wait for IAM policies to propagate",
+    )
+    args = parser.parse_args()
+
+    # Set up.
+    ctx = TestContext(iam_propagation_seconds=args.iam_propagation_seconds)
+
+    # Create the "jump role" that Materialize will use to assume each
+    # connection's role.
+    connection_role = f"testdrive-{ctx.seed}-MaterializeConnection"
+    connection_role_arn = f"arn:aws:iam::{ctx.account_id}:role/{connection_role}"
+    ctx.iam.create_role(
+        RoleName=connection_role,
+        AssumeRolePolicyDocument=json.dumps(
+            {
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Effect": "Allow",
+                        "Principal": {
+                            "AWS": ctx.materialized_principal,
+                        },
+                        "Action": "sts:AssumeRole",
+                    }
+                ],
+            }
+        ),
+    )
+
+    # Start Materialize.
+    materialized = Materialized(
+        environment_extra=[
+            "AWS_DEFAULT_REGION=us-east-1",
+            "AWS_ACCESS_KEY_ID",
+            "AWS_PROFILE",
+            "AWS_SECRET_ACCESS_KEY",
+            "AWS_SESSION_TOKEN",
+        ],
+        volumes_extra=[
+            # Mounting the .aws directory in the container allows Materialize to
+            # use SSO credentials, which makes it easier to run this composition
+            # locally. CI doesn't need this.
+            "~/.aws:/home/materialize/.aws",
+        ],
+        options=[
+            f"--aws-connection-role-arn={connection_role_arn}",
+            f"--aws-external-id-prefix={AWS_EXTERNAL_ID_PREFIX}",
+        ],
+    )
+    with c.override(materialized):
+        # (Re)start Materialize and enable AWS connections.
+        c.down()
+        c.up("materialized")
+        c.sql(
+            port=6877,
+            user="mz_system",
+            sql="""
+            ALTER SYSTEM SET enable_aws_connection = true;
+            ALTER SYSTEM SET enable_connection_validation_syntax = true;
+            """,
+        )
+
+        for fn in [test_credentials, test_assume_role]:
+            with c.test_case(fn.__name__):
+                fn(c, ctx)
+
+
+def test_credentials(c: Composition, ctx: TestContext):
+    # Create a user with an access key.
+    customer_user = f"testdrive-{ctx.seed}-Customer"
```

Comment:
Should the users and roles created in this test be deleted as well as a cleanup after the test? If I am understanding right, these are actually created in Materialize Scratch account whenever they are run at least locally and we'll someday hit the limit?

<img width="682" alt="image" src="https://github.com/MaterializeInc/materialize/assets/2379129/6936f2b6-b28d-4a7a-86cb-e9bf13c26a24">